keras/models.py

```diff
@@ -81,6 +81,8 @@ def weighted(y_true, y_pred, weights, mask=None):
         # score_array has ndim >= 2
         score_array = fn(y_true, y_pred)
         if mask is not None:
+            # Cast the mask to floatX to avoid float64 upcasting in theano
+            mask = K.cast(mask, K.floatx())
```

Comment:
Well I'm not an expert of keras nor theano. I also sent an email to theano-users list today but didn't receive anything yet: On my system a multiplication of float32 and int32 in theano results in creation of a float64 tensor. Here the mask is int32 and the score_array is float32. 

Are casts causing transfer between host and device?